### PR TITLE
Add sdk information in headers for various services

### PR DESCRIPTION
**Pull request by joaquim-verges**
## Problem solved

We dont have any data on which SDK versions are using our services currently.

## Changes made

- [x] Public API changes: none
- [x] Internal API changes: added `x-sdk` headers to rpc/storage/bundler calls

## How to test

- [x] Manual tests: inspect request headers

### CODE REVIEW

**Review by joaquim-verges — COMMENTED** [review]
Thank you @MananTank !

File: packages/wallets/src/evm/connectors/smart-wallet/lib/http-rpc-client.ts [hunk 1]
```diff
@@ -2,8 +2,12 @@ import { providers, utils } from "ethers";
 import { UserOperationStruct } from "@account-abstraction/contracts";
 import { deepHexlify } from "@account-abstraction/utils";
 import { isTwUrl } from "../../../utils/url";
+import pkg from "../../../../../package.json";
 
 const DEBUG = false;
+function isBrowser() {
+  return typeof window !== "undefined";
```
Comment (iketw) [inline packages/wallets/src/evm/connectors/smart-wallet/lib/http-rpc-client.ts:9]: This is too simple imo, platforms like ReactNative will also return true here. I recently discovered that this is a thing:

`global?.navigator?.product === 'ReactNative'`, so we can use this to differentiate react-native, browser, node

File: packages/storage/tsconfig.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-  "extends": "@thirdweb-dev/tsconfig/base.json",
+  "extends": "@thirdweb-dev/tsconfig/sdk.json",
```
[the remark this answers is not in the document]
Comment (joaquim-verges) [reply packages/storage/tsconfig.json:2]: Needed the Json resolution flag

File: packages/wallets/src/evm/connectors/smart-wallet/lib/http-rpc-client.ts (hunk 1 above)
Comment (joaquim-verges) [reply packages/wallets/src/evm/connectors/smart-wallet/lib/http-rpc-client.ts:9]: Ooh that's nice, can you add it for me?

Def want to differentiate reqs from react native, thanks for catching that

Comment (joaquim-verges) [reply packages/wallets/src/evm/connectors/smart-wallet/lib/http-rpc-client.ts:9]: hmmm that gives me weird names in browser though. I think i'll just consider react native if we have a bundle id

Comment (iketw) [reply packages/wallets/src/evm/connectors/smart-wallet/lib/http-rpc-client.ts:9]: Yeah, looking deeper it seems that the feature is deprecated: 

https://developer.mozilla.org/en-US/docs/Web/API/Navigator/product

Comment (iketw) [reply packages/wallets/src/evm/connectors/smart-wallet/lib/http-rpc-client.ts:9]: It seems we can do something like this though:

```
function getEnvironment() {
  if (typeof navigator !== 'undefined' && navigator.product === 'ReactNative') {
    return 'React Native';
  } else if (typeof window !== 'undefined') {
    return 'React';
  } else {
    return 'Unknown';
  }
}
```

**Pull request merged**